library/src/blas2/rocblas_spr.hpp

```diff
@@ -0,0 +1,114 @@
+/* ************************************************************************
+ * Copyright 2016-2020 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#pragma once
+#include "handle.h"
+#include "rocblas.h"
+
+template <typename T>
+__device__ void
+    spr_kernel_calc(bool upper, rocblas_int n, T alpha, const T* x, rocblas_int incx, T* AP)
+{
+    rocblas_int tx = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    rocblas_int ty = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+    int index = upper ? ((ty * (ty + 1)) / 2) + tx : ((ty * (2 * n - ty + 1)) / 2) + (tx - ty);
+    if(index < 0 || index > (n * (n + 1)) / 2)
```

Comment:
index = (n(n+1))/2 is okay?

Reply:
Looks like that check is pointless as the tx and ty checks ensure that index is indeed in that range (and exclusively less than n(n+1)/2)